diff --git a/libavformat/movenc.c b/libavformat/movenc.c
index 5608afde42..6edef4b5c4 100644
--- a/libavformat/movenc.c
+++ b/libavformat/movenc.c
@@ -4380,18 +4380,19 @@ static int mov_write_uuidusmt_tag(AVIOContext *pb, AVFormatContext *s)
     return 0;
 }
 
-static void build_chunks(MOVTrack *trk)
+static void build_chunks(AVFormatContext* s, MOVTrack *trk)
 {
     int i;
     MOVIentry *chunk = &trk->cluster[0];
     uint64_t chunkSize = chunk->size;
+    int maxChunkSize = (s->max_chunk_size > 0) ? s->max_chunk_size : (1<<20);
     chunk->chunkNum = 1;
     if (trk->chunkCount)
         return;
     trk->chunkCount = 1;
     for (i = 1; i<trk->entry; i++){
         if (chunk->pos + chunkSize == trk->cluster[i].pos &&
-            chunkSize + trk->cluster[i].size < (1<<20)){
+            chunkSize + trk->cluster[i].size < maxChunkSize){
             chunkSize             += trk->cluster[i].size;
             chunk->samples_in_chunk += trk->cluster[i].entries;
         } else {
@@ -4466,7 +4467,7 @@ static int mov_write_moov_tag(AVIOContext *pb, MOVMuxContext *mov,
         mov->tracks[i].time     = mov->time;
 
         if (mov->tracks[i].entry)
-            build_chunks(&mov->tracks[i]);
+            build_chunks(s, &mov->tracks[i]);
     }
 
     if (mov->chapter_track)
